sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -393,3 +393,71 @@ case class CreateNamedStructUnsafe(children: Seq[Expression]) extends Expression
 
   override def prettyName: String = "named_struct_unsafe"
 }
+
+/**
+ * Creates a map after splitting the input text into key/value pairs using delimeters
+ */
+@ExpressionDescription(
+  usage = """_FUNC_(text[, pairDelim, keyValueDelim]) - Creates a map after splitting the text into
+    key/value pairs using delimiters.
+    Default delimiters are ',' for pairDelim and '=' for keyValueDelim.""")
+case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: Expression)
+  extends TernaryExpression with ExpectsInputTypes {
+
+  def this(child: Expression) = {
+    this(child, Literal(","), Literal("="))
+  }
+
+  override def children: Seq[Expression] = Seq(text, pairDelim, keyValueDelim)
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType, StringType)
+
+  override def dataType: DataType = MapType(StringType, StringType, valueContainsNull = false)
+
+  override def nullSafeEval(str: Any, delim1: Any, delim2: Any): Any = {
+    val array = str.asInstanceOf[UTF8String]
+      .split(delim1.asInstanceOf[UTF8String], -1)
+      .map{_.split(delim2.asInstanceOf[UTF8String], 2)}
+
+    ArrayBasedMapData(array.map(_(0)), array.map(_(1))).asInstanceOf[MapData]
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+    nullSafeCodeGen(ctx, ev, (text, delim1, delim2) => {
+      val arrayClass = classOf[GenericArrayData].getName
+      val mapClass = classOf[ArrayBasedMapData].getName
+      val keyArray = ctx.freshName("keyArray")
+      val valueArray = ctx.freshName("valueArray")
+      ctx.addMutableState("UTF8String[]", keyArray, s"this.$keyArray = null;")
+      ctx.addMutableState("UTF8String[]", valueArray, s"this.$valueArray = null;")
```

Comment:
It's used for split the code into some small blocks. BTW why would you do `this.keyArray = null;`? We can declare `keyArray` as local variable.